dblookupext/historyRepository.go

```diff
@@ -383,112 +240,51 @@ func (hr *historyRepository) onNotarizedMiniblock(metaBlockNonce uint64, metaBlo
-		metadata.NotarizedAtSourceInMetaNonce = notification.metaNonce
-		metadata.NotarizedAtSourceInMetaHash = notification.metaHash
-	})
-
-	hr.consumePendingNotificationsNoLock(hr.pendingNotarizedAtDestinationNotifications, func(metadata *MiniblockMetadata, notification *notarizedNotification) {
-		metadata.NotarizedAtDestinationInMetaNonce = notification.metaNonce
-		metadata.NotarizedAtDestinationInMetaHash = notification.metaHash
-	})
-
-	hr.consumePendingNotificationsNoLock(hr.pendingNotarizedAtBothNotifications, func(metadata *MiniblockMetadata, notification *notarizedNotification) {
-		metadata.NotarizedAtSourceInMetaNonce = notification.metaNonce
-		metadata.NotarizedAtSourceInMetaHash = notification.metaHash
-		metadata.NotarizedAtDestinationInMetaNonce = notification.metaNonce
-		metadata.NotarizedAtDestinationInMetaHash = notification.metaHash
-	})
-
-	log.Trace("consumePendingNotificationsWithLock() end",
-		"len(source)", hr.pendingNotarizedAtSourceNotifications.Len(),
-		"len(destination)", hr.pendingNotarizedAtDestinationNotifications.Len(),
-		"len(both)", hr.pendingNotarizedAtBothNotifications.Len(),
-	)
-}
-
-func (hr *historyRepository) consumePendingNotificationsNoLock(pendingMap *container.MutexMap, patchMetadataFunc func(*MiniblockMetadata, *notarizedNotification)) {
-	for _, key := range pendingMap.Keys() {
-		notification, ok := pendingMap.Get(key)
-		if !ok {
-			continue
+	if isNotarizedAtSource {
+		updateHandler := func(mbMetadataOnBlock *MiniblockMetadataOnBlock) {
+			mbMetadataOnBlock.NotarizedAtSourceInMetaHash = metaBlockHash
+			mbMetadataOnBlock.NotarizedAtSourceInMetaNonce = metaBlockNonce
 		}
 
-		keyTyped, ok := key.(string)
-		if !ok {
-			log.Error("consumePendingNotificationsNoLock(): bad key", "key", key)
-			continue
+		err := hr.miniblocksHandler.updateMiniblockMetadataOnBlock(miniblockHash, headerHash, updateHandler)
+		if err != nil {
+			log.Warn("historyRepository.onNotarizedMiniblock - isNotarizedAtSource", "error", err)
 		}
 
-		notificationTyped, ok := notification.(*notarizedNotification)
-		if !ok {
-			log.Error("consumePendingNotificationsNoLock(): bad value", "value", fmt.Sprintf("%T", notification))
-			continue
-		}
+		return
+	}
 
-		miniblockHash := []byte(keyTyped)
-		metadata, err := hr.getMiniblockMetadataByMiniblockHash(miniblockHash)
-		if err != nil {
-			// Maybe not yet committed / saved in storer
-			continue
+	if isNotarizedAtDestination {
+		updateHandler := func(mbMetadataOnBlock *MiniblockMetadataOnBlock) {
+			mbMetadataOnBlock.NotarizedAtDestinationInMetaHash = metaBlockHash
+			mbMetadataOnBlock.NotarizedAtDestinationInMetaNonce = metaBlockNonce
 		}
 
-		patchMetadataFunc(metadata, notificationTyped)
-		err = hr.putMiniblockMetadata(miniblockHash, metadata)
+		err := hr.miniblocksHandler.updateMiniblockMetadataOnBlock(miniblockHash, headerHash, updateHandler)
```

Comment:
I think that this code can be extracted outside the `if` clauses and called only once with the obtained `updateHandler`:
```
err := hr.miniblocksHandler.updateMiniblockMetadataOnBlock(miniblockHash, headerHash, updateHandler)
if err != nil {
    ...
}
```

Reply:
refactored code